.dotnet/src/Custom/OpenAIClientConnector.cs

```diff
@@ -22,7 +20,6 @@ internal partial class OpenAIClientConnector
         ApiKeyCredential credential = null,
         OpenAIClientOptions options = null)
     {
-        if (model is null) throw new ArgumentNullException(nameof(model));
         Model = model;
         Endpoint ??= options?.Endpoint ?? new(Environment.GetEnvironmentVariable(s_OpenAIEndpointEnvironmentVariable) ?? s_defaultOpenAIV1Endpoint);
         credential ??= new(Environment.GetEnvironmentVariable(s_OpenAIApiKeyEnvironmentVariable) ?? string.Empty);
```

Comment:
I think we should remove this call to GetEnvironmentVariable until we have proper support for this via IConfig-like solution.

Reply:
I'd like to learn more about the concerns with having implicit env-variable-based initialization -- let's follow up separately to this PR since that can stand as its own good topic.